### PR TITLE
add asciidoctor support

### PR DESCRIPTION
**Pull request by kwark**
Added asciidoctor support including scripted test
Credit go to https://github.com/kodemaniak/sbt-site/ which had this code in his fork
I just updated to use asciidoctorj and added the scripted test

I set the version explicitly to 0.8.2-SNAPSHOT in order to test with scripted

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @kwark,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://typesafe.com/contribute/cla](http://typesafe.com/contribute/cla)

**Comment by kwark** [comment]
I accepted the Typesafe CLA.
Is there anybody here to review this PR?

**Comment by ktoso** [comment]
I'm not the maintainer of this one (but have an WIP asciidoctor plugin as well https://github.com/ktoso/asciidoctor-sbt-plugin ). Looks good from my point of view :)
Let's also ping @kodemaniak for kudos and informing if he's OK for this to be pulled in here in such way.

**Comment by kodemaniak** [comment]
Actually, I totally forgot that I had this fork :). But I would really love to see asciidoctor support in sbt-site, so please continue. I don't remember why I did not proceed with my version.

File: build.sbt [hunk 1]
```diff
@@ -8,7 +8,8 @@ organization := "com.typesafe.sbt"
 
 scalaVersion := "2.10.4"
 
-versionWithGit
+//versionWithGit
+version := "0.8.2-SNAPSHOT"
```
Comment (jsuereth) [inline build.sbt]: can you undo this before we merge?

File: src/main/scala/com/typesafe/sbt/site/AsciidoctorSupport.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      target in Asciidoctor <<= target(_ / "asciidoctor"),
+      includeFilter in Asciidoctor := AllPassFilter) ++ inConfig(Asciidoctor)(Seq(
+      mappings <<= (sourceDirectory, target, includeFilter, version) map AsciidoctorRunner.run))
+}
+
+object AsciidoctorRunner {
+
+  def run(input: File, output: File, includeFilter: FileFilter, version: String): Seq[(File, String)] = {
+    val oldContextClassLoader = Thread.currentThread().getContextClassLoader
+    Thread.currentThread().setContextClassLoader(this.getClass.getClassLoader)
```
Comment (jsuereth) [inline src/main/scala/com/typesafe/sbt/site/AsciidoctorSupport.scala]: Man, we really need this as a utility in the standard library or something :)

**Comment by jsuereth** [comment]
LGTM, caveat one comment on the build.

**Comment by kwark** [comment]
@jsuereth updated the build.sbt as requested
Thanks for taking care of this PR

**Comment by jsuereth** [comment]
Thanks for the fix and sorry it took me so long to get back to this!

**Pull request merged**